packages/graphql/src/scripts/201904_backfill_avatar.js

```diff
@@ -39,67 +44,89 @@ async function backfillAvatarToIpfs() {
     }
   })
 
+  const tasks = []
   for (const account in accountIpfs) {
     for (const identityIpfsHash of accountIpfs[account]) {
-      console.log(
-        'Starting Profile',
-        originIpfs.getIpfsHashFromBytes32(identityIpfsHash)
-      )
-      console.log(' for account', account)
-      const data = await originIpfs.get(contracts.ipfsGateway, identityIpfsHash)
-      if (!data) {
-        showStatus('FAIL: Failed to fetch identity IPFS data')
-        continue
-      }
-      if (data.profile === undefined) {
-        showStatus('SKIP: No profile')
-        continue
-      }
-      if (data.profile.avatarUrl) {
-        showStatus('SKIP: Already using new avatarUrl')
-        continue
-      }
-      if (data.profile.avatar === undefined || data.profile.avatar === '') {
-        showStatus('SKIP: No avatar')
-        continue
-      }
-      if (!data.profile.avatar.startsWith('data:')) {
-        showStatus('SKIP: Not data url for avatar')
-        continue
-      }
-      const { buffer, mimeType } = dataURItoBinary(data.profile.avatar)
-      if (buffer.length == 0) {
-        showStatus('SKIP: Not actual data in data url for avatar')
-        continue
-      }
-      const calculatedUrl = await IpfsHash.of(Buffer.from(buffer))
-      console.log('', mimeType, buffer.length)
-      const actualUrl = await postFile(contracts.ipfsRPC, buffer, mimeType)
-      console.log('', 'saved as ', `${contracts.ipfsGateway}/ipfs/${actualUrl}`)
-
-      if (actualUrl != calculatedUrl) {
-        showStatus('FAIL: Calculated IPFS does not match upload')
-        console.error(calculatedUrl, actualUrl)
+      if (ONLY_ACCOUNT !== undefined && ONLY_ACCOUNT != account) {
         continue
       }
-      showStatus('SUCCESS: Avatar uploaded to IPFS')
+      tasks.push({ account, identityIpfsHash })
     }
   }
+
+  for (const chunklet of chunk(tasks, CONCURRENCY)) {
```

Comment:
This totally works. An alternative that might be simpler FYI is to use the p-limit package.
See https://github.com/OriginProtocol/origin/blob/master/infra/discovery/src/listener/listener.js#L173